joblib/test/test_parallel.py

```diff
@@ -940,3 +942,43 @@ def __reduce__(self):
     with warns(DeprecationWarning):
         with raises(ValueError):
             delayed(UnpicklableCallable(), check_pickle=True)
+
+
+@with_multiprocessing
+@parametrize('backend', ['multiprocessing', 'loky'])
+def test_backend_batch_statistics_reset(backend):
+    """Test that parallel backend correctly resets its batch statistics."""
+
+    # let's tolerate 20% error between compuration times to avoid random
+    # failures because of variable available ressources on CIs.
+    error = 0.2
+    n_jobs = 2
+    n_inputs = 100
+    task_time = 2. / n_inputs
+
+    # Use default backend
+    p = Parallel(verbose=10, n_jobs=n_jobs, backend=backend)
+    start_time = time.time()
+    p(delayed(time.sleep)(task_time) for i in range(n_inputs))
+    ref_time = time.time() - start_time
+    assert p._backend._effective_batch_size == DEFAULT_EFFECTIVE_BATCH_SIZE
+    assert (p._backend._smoothed_batch_duration ==
+            DEFAULT_SMOOTHED_BATCH_DURATION)
+
+    start_time = time.time()
+    p(delayed(time.sleep)(task_time) for i in range(n_inputs))
+    test_time = time.time() - start_time
+    assert p._backend._effective_batch_size == DEFAULT_EFFECTIVE_BATCH_SIZE
+    assert (p._backend._smoothed_batch_duration ==
+            DEFAULT_SMOOTHED_BATCH_DURATION)
+
+    assert (
+        # With Loky, the test run is a lot faster than the reference
+        # run because it reuses previously created workers.
+        # Having test_time smaller than reference time is ok for both
+        # backends.
+        test_time < ref_time or
```

Comment:
rather than a single condition for both backends you can have a condition per backend

```py
if backend == 'multiprocessing':
    assert abs(...)
elif backend == 'loky':
    assert test_time < ref_time
else:
    raise ValueError('You need to add an assert statement for backend {}'.format(backend))
```